### ISSUE TITLE
Updating the outcome of a proposal after it is closed

### ISSUE THREAD

**Issue by niceking** [loomio/loomio#2120]
1. After a proposal had closed and I had published an outcome, a month later I tried to change the outcome of the proposal i made 
2. Page refreshed, got a flash message about the outcome being updated, however I scrolled down and the outcome was now empty
3. Repasted in my text and tried to publish the outcome again, I *think* i got a slightly different success message saying that the outcome has been published. This time text was persisted
4. Did this email everyone who responded? Unclear to me 
5. Have not tried to replicate in case it emails everyone each time

Attached is a screenshot showing that the outcome was updated, and the republished. Also a link to the loomio for those with access to this group 

https://www.loomio.org/d/zL4cBXu1/rails-girls-at-enspiral-space?proposal=8dGKF3iO

![screen shot 2015-04-22 at 12 13 46 pm](https://cloud.githubusercontent.com/assets/4241125/7265147/55ead446-e8e9-11e4-87bf-8a7ada866084.png)

**Comment by zacharystenger** [comment]
I was able to replicate and fix this bug on my development environment.  
In the MotionsController, the update_outcome method makes this call 
```ruby
if MotionService.update_outcome(motion: @motion, params: params, actor: current_user) 
```
which passes params that look something like this (scroll to middle to see motion & outcome objects):
```ruby
{"utf8"=>"✓", "_method"=>"patch", "authenticity_token"=>"XvDLKFeyGSBoYn+FYMeiL2oXinKUndQYJIUmaegZq0pBrng5TjkXXmom+NeN2ftvow0JUhVydxBvffaXL1B/pg==", "motion"=>{"outcome"=>"updated outcome goes here"}, "commit"=>"Publish outcome", "controller"=>"motions", "action"=>"update_outcome", "id"=>"fTcLrP4k"}
```
The problem is the outcome object is inside the motion object so I changed this to pass permitted_params.motion instead like it does in the create_outcome method
```ruby
if MotionService.update_outcome(motion: @motion, params: permitted_params.motion, actor: current_user)
```
which passes params that look something like this:
```ruby
{"outcome"=>"updated outcome goes here"}
```
And it worked (as far as I can tell, but I'm new to Loomio...)  

Another possible fix would be to change the MotionService update_outcome instead. Currently it does this:
```ruby
motion.outcome = params[:outcome]
```
But this sets the outcome to nil because params doesn't immediately have an outcome object, it is within the motion object.  To fix it change that line to this:  
```ruby
motion.outcome = params[:motion][:outcome]
```
I have also confirmed that this solution works but I believe my other solution is better since it keeps the structure of create_outcome and update_outcome more similar and no longer passes unnecessary params.

I would like to submit a PR for my first solution (changing params to permitted_params.motion) but wanted some feedback first.  
Thanks

**Comment by gdpelican** [comment]
Hey @zacharystenger, thanks for taking a peek at this.

Changing `params: params` to `params: permitted_params.motion` is the way to go here; you'll note that we do it this way throughout the application, including `MotionController#create` and `MotionController#update`, so this seems to have been an oversight.

I'd be super happy to accept a PR for this.

Thanks again! :ribbon:

**Comment by gdpelican** [comment]
Fixed in #2203
